test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperParseTreeRewriterTest.cs

```diff
@@ -19,6 +19,832 @@ namespace Microsoft.AspNet.Razor.Test.TagHelpers
+                            new MarkupBlock(
+                                factory.EmptyHtml(),
+                                new StatementBlock(
+                                    factory.CodeTransition(),
+                                    factory.MetaCode("{").Accepts(AcceptedCharacters.None),
+                                    new MarkupBlock(
+                                        new MarkupTagHelperBlock("p",
+                                            new MarkupTagHelperBlock("strong",
+                                                blockFactory.EscapedMarkupTagBlock("</", "strong>", AcceptedCharacters.None)))),
+                                    new MarkupBlock(
+                                        blockFactory.EscapedMarkupTagBlock("<", "p>", AcceptedCharacters.None),
+                                        blockFactory.MarkupTagBlock("</strong>", AcceptedCharacters.None)),
+                                    new MarkupBlock(
+                                        blockFactory.EscapedMarkupTagBlock("</", "p>", AcceptedCharacters.None)),
+                                    factory.EmptyCSharp().AsStatement(),
+                                    factory.MetaCode("}").Accepts(AcceptedCharacters.None)),
+                                factory.EmptyHtml()),
+                        new []
+                        {
+                            new RazorError(
+                                string.Format(errorFormatNormalUnclosed, "p", CultureInfo.InvariantCulture),
+                                absoluteIndex: 2, lineIndex: 0, columnIndex: 2),
+                            new RazorError(
+                                string.Format(errorFormatMalformed, "p", CultureInfo.InvariantCulture),
+                                absoluteIndex: 2, lineIndex: 0, columnIndex: 2),
+                            new RazorError(
+                                string.Format(errorFormatMalformed, "strong", CultureInfo.InvariantCulture),
+                                absoluteIndex: 5, lineIndex: 0, columnIndex: 5),
+                            new RazorError(
+                                string.Format(errorFormatNormalUnclosed, "!p", CultureInfo.InvariantCulture),
+                                absoluteIndex: 23, lineIndex: 0, columnIndex: 23),
+                            new RazorError(
+                                string.Format(errorFormatMalformed, "strong", CultureInfo.InvariantCulture),
+                                absoluteIndex: 27, lineIndex: 0, columnIndex: 27),
+                            new RazorError(
+                                string.Format(errorFormatNormalNotStarted, "!p", CultureInfo.InvariantCulture),
+                                absoluteIndex: 36, lineIndex: 0, columnIndex: 36),
+                        }
+                    },
+                };
+            }
+        }
+
+        public static TheoryData OptOutAttributeDataCSharp
```

Comment:
nit: reads like an attribute is opting out, suggest `OptOut_WithAttributeData_CSharp`